src/xhydro/pmp.py

```diff
@@ -0,0 +1,590 @@
+"""Module to compute Probable Maximum Precipitation."""
+
+import sys
+from itertools import product
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+import xclim
+from xclim.indices.stats import fit, parametric_quantile
+
+
+def major_precipitation_events(da, acc_day, quantil=0.9, path=None):
+    """Extract precipitation events that exceed a given quantile for a given days accumulation.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+        DataArray containing the precipitation values.
+    acc_day : list
+        List of precipitation accumulation days.
+    quantil : float
+        Threshold that limits the events to those that exceed this quantile.
+    path : str, optional
+        Path where the results will be saved.
+
+    Returns
+    -------
+    xr.DataArray
+        Precipitation events that exceed the given quantile for
+        the given days accumulation.
+    """
+    if "y" and "x" in da.coords:
+        core_dims = ["time", "y", "x"]
+
+    if "conf" in da.coords:
+        core_dims = ["time", "conf"]
+
+    ds_exp = da.expand_dims(dim={"acc_day": acc_day}).copy(deep=True)
+    da_acc = xr.apply_ufunc(
+        cumulate_precip,
+        ds_exp,
+        acc_day,
+        input_core_dims=[core_dims, []],
+        output_core_dims=[core_dims],
+        dask="parallelized",
+        output_dtypes=[float],
+        vectorize=True,
+    )
```

Comment:
I created the cumulate_precip and used it along with the xr.apply_ufunc() because I thought that was gonna be more efficient. What I understand is that using xr.apply_ufunc()  dask can parallelized the the function along the dimensions, including the accumulation days, thus avoiding the loop and the concatenation. Is this true? I don't have much experience working with xarrays.

Reply:
You are right @ospinajulian about speeding up calculation with `xr.apply_ufunc()` BUT for some functions, Xarray already provides the benefits of `xr.apply_ufunc()` + some other optimizations. Hence, in a Dask environment, the code provided by @RondeauG would work with Dask out of the box and thus be parallelized.

Reply:
got it !